include/hipSYCL/std/stdpar/detail/algorithm_fwd.hpp

```diff
@@ -55,6 +55,11 @@ HIPSYCL_STDPAR_ENTRYPOINT ForwardIt2 copy_n(hipsycl::stdpar::par_unseq,
                                             ForwardIt1 first, Size count,
                                             ForwardIt2 result);
 
+template <class ForwardIt1, class ForwardIt2>
```

Comment:
Why are only the `par_unseq` policy variants forward declared?

Reply:
Ohhh... good catch! Even for `par_unseq`, it's incomplete, as some newer algorithms like `scan` or `merge` are missing...I think at some point probably I just forgot to add the forward declarations :/

They all should be forward-declared (for both `par` and `par_unseq`). The actual implementations are pulled in by `execution`, so those forward declares might only matter in specific cases (perhaps depending on the order in which `execution` and `algorithm`/`numeric` are included?).

Reply:
Thanks for the explanation! I'll forward declare both.